### PR TITLE
Voice: send feedback to an item when listening for a command

### PR DESCRIPTION
**Pull request by ghys**
Implments #3450.

Signed-off-by: Yannick Schaus <eclipse@schaus.net>

### CODE REVIEW

**Comment by kubawolanin** [comment]
Nice!
Is there any timeout in place, so the user knows when the Listening switch goes back to `OFF`?

File: bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java [hunk 1]
```diff
@@ -81,28 +88,42 @@ public DialogProcessor(KSService ks, STTService stt, TTSService tts, HumanLangua
...
-            processing = true;
             this.isSTTServerAborting = false;
             if (ksEvent instanceof KSpottedEvent) {
+                toggleProcessing(true);
                 if (stt != null) {
                     try {
                         this.sttServiceHandle = stt.recognize(this, source.getInputStream(format), this.locale,
                                 new HashSet<String>());
                     } catch (STTException | AudioException e) {
+                        logger.error("Error during recognition", e);
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java]: Is this really a "software bug" situation which requires error logging and a stack trace?

Comment (ghys) [inline bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java]: Actually I had initially an exception coming from the `source.getInputStream(format)` call (wrong capture frequency IIRC). So I only had the message "Error creating the audio input stream" without the cause. Maybe it makes sense to only log an `AudioException`?

**Comment by ghys** [comment]
> Is there any timeout in place, so the user knows when the Listening switch goes back to OFF?

I thought it was better not to handle a timeout here but let the `STTService` do it itself. It simply needs to send a `RecognitionStopEvent` and the switch will be set back to OFF (meaning it's in the keyword spotting state again).

File: bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java (hunk 1 above)
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/DialogProcessor.java]: > Maybe it makes sense to only log an AudioException?

Yes, I think so, as it is a unchecked exception, while STTException is a checked one.

**Comment by kaikreuzer** [comment]
@ghys Could you update this PR regarding the logging/exception handling? We would be good to merge then.

**Comment by ghys** [comment]
@kaikreuzer sure (was AFK for a few days) - done!

**Pull request merged**